server/e2e_test.go

```diff
@@ -214,6 +214,30 @@ func TestPassivePlayersGetBanEffects(t *testing.T) {
 	h.WaitEmptyInboxes(allClients)
 }
 
+func TestDirectMessageFromDisconnectedUserIsIgnored(t *testing.T) {
```

Comment:
This can be removed as we don't need this logic. Also gets rid of the sleep in the tests. ;)

Reply:
Do you mean you want to remove this logic?

```
sendingPlayer := pi.tracker.playerByConnection(pi.conn)
if sendingPlayer == nil {
	if debugMode {
		fmt.Printf("[DirectMessage] Ignoring from %s because player no longer exists\n", getIP(pi.conn))
	}
	return
}
```

Reply:
(as well as the test that you are commenting on here)

Reply:
Yep

Reply:
Ah right! Many thanks for catching this. We do want to send messages even after disconnect.

I needed to add a frozen pool lookup so that the broadcast can also allow messages from disconnected users (direct message work without the user but the broadcast does not since there is no way to find the pool).

I confirmed that the lookup works with a new test.

On a related note, this "send + disconnect" behavior actually had a race condition. I will describe it in another message.